Is-884: add dbtool

inwonkim opened the pull request:
* Enable shorten option
* precommit can be queried with blockHeight or address or both of them
* accountDB can be queried in 4  ways
      - with path option
      - with path, address option
      - with rcRoot option, accountType option
      - with rcRoot option, accountType, address  option

Commit:
Add dbtool querying rc, iiss data

* Query data with given path of db
* Need rcRootPath when querying specified address in accountDB

Commit:
Improve dbtool

Commit:
Write account querying command and precommit querying command

* precommit can be queried with blockHeight or address
* accountDB can be queried in 4  ways
  - with path option
  - with path, address option
  - with rcRoot option, accountType option
  - with rcRoot option, accountType, address  option

eunsoo-icon commented on cmd/dbtool/db_account.go:
```diff
@@ -0,0 +1,211 @@
+	if input.path != "" {
+		queryAccountWithAccountDBPath(input.path, input.address)
+		return
+	} else if !(input.rcDBRoot == "" && input.accountType == "") {
```
Can't handle command if set only -type options without -dbroot

Commit:
refactor dbtool

Commit:
Refactor dbtool

* Modify iteratePrintDB to get Entries and change its name
* Print entries in printEntries function
* Write runQuery0000 functions
  - The runQuery000 functions return an instance of the object you were
  querying for.
* Fix input validation

Commit:
Rollback iteratePrintDB. Enable 4 combinations of options for querying account db
 - account == "", accountType == ""
 - account == "", accountType != ""
 - account != "", accountType == ""
 - account != "", accountType != ""

Commit:
Use printAllEntriesInPath in db_000.go files. fix option validation for
querying account db

Commit:
Refactor dbtool

- query0000 functions return err and handle error in main() instance of instantly call os.Exit()
- rename functions

Commit:
IS-884: add prep type to manage
